### DISCUSSION TITLE
Contribution Guide

### DISCUSSION THREAD

**Discussion by ShaMan123** [fabricjs/fabric.js#7167] [Ideas]
I think the contribution guide should be visible both in the README and in the official site.
I think it should include more information on the dev workflow, some is in the wiki, some is in `CONTRIBUTING.md`.
- PR essentials
   - update CHANGELOG
   - lint
   - create demo?
- Setting up the development environment
- Overviews on tests, site, useful commands, build process etc.

I would like this discussion to include all relevant information and points so a **Contribution Guide PR** can be made.

@asturur @melchiar

**Comment by ShaMan123** [comment]
https://github.com/fabricjs/fabricjs.com/blob/gh-pages/posts/demos/_posts/2020-2-15-custom-control-render.md

This is an example of the demo.
You create a new file in the same folder, and start writing.

To load the website locally:
you clone the repo in the same parent folder of fabricJS, so that fabric.js and fabricjs.com are siblings.
Then in fabricjs.com folder you run `jekyll serve`.
If you are on windows, i have no idea how to exactly have it running. It works well on mac and linux.

_Originally posted by @asturur in https://github.com/fabricjs/fabric.js/issues/6994#issuecomment-845435948_

**Reply by ShaMan123** [reply]
https://github.com/fabricjs/fabric.js/wiki/How-to-contribute-to-Fabric

**Reply by asturur** [reply]
It should clearly have a way to give something to do to people that come with the will to contribute but do not have a task in mind, and for whoever comes with a ready idea how to better present it.

**Comment by asturur** [comment]
It would be great to improve it.
Should be avaialable as contributing.MD i think as in other repositories right?

**Reply by ShaMan123** [reply]
Absolutely!
IMO the how-to is very important. If someone wants to contribute it should be easy to understand how to set up the environment etc. For me it wasn't.

**Comment by ShaMan123** [comment]
What about this CI for PRs?
https://codesandbox.io/docs/ci

**Reply by asturur** [reply]
github actions seems fine free and fast.
i had travis CI before and had issues when FOSS wasn't free anymore.
Why would you switch to codesandbox?

**Reply by ShaMan123** [reply]
> 1. I prefer codesandbox because it is an actual project (structure, config, files and all) and flexibility as opposed to codepen which has only 3 files AFAIK (html, js, css). I am willing to support an option to export to codepen but I need guidance.
>    
>    * Is it possible to add more files?
>    * If not how should I squash all the code into the js file?

> _Originally posted by @ShaMan123 in https://github.com/fabricjs/fabric.js/issues/7178#issuecomment-877843077_

**Reply by asturur** [reply]
Sorry i misunderstood your question.
I thought that codesandbox CI was a full CI system, and not a plugin for github actions.
We can add that too, i m not opposed.

**Comment by ShaMan123** [comment]
#7178
